numpyro/contrib/nested_sampling.py

```diff
@@ -245,8 +233,17 @@ def run(self, rng_key, *args, **kwargs):
         else:
             log_density_ = log_density
 
-        def loglik_fn(**params):
-            return log_density_(reparam_model, args, kwargs, params)[0]
+        # Jaxns requires loglikelihood function to have explicit signatures.
+        local_dict = {}
+        loglik_fn_def = """def loglik_fn({}):\n
+        \tparams = dict({})\n
+        \treturn log_density_(reparam_model, args, kwargs, params)[0]
+        """.format(
+            ", ".join([f"{name}_base" for name in param_names]),
+            ", ".join([f"{name}_base={name}_base" for name in param_names]),
+        )
+        exec(loglik_fn_def, locals(), local_dict)
```

Comment:
oof, I guess it's not worth introducing a dependency on [makefun](https://pypi.org/project/makefun/), which is also used by funsor

Reply:
Yeah using `makefun` would make the code much simpler. Thanks for pointing out!